### PR TITLE
chore: update tx history

### PR DESCRIPTION
**Pull request by catngh**
**What does this PR do?**

- New API `/offchain-tip-bot/history` params: `guild_id`, `token`
image

### CODE REVIEW

File: pkg/entities/offchain_tip_bot_transfer_histories.go [hunk 1]
```diff
@@ -11,3 +12,12 @@ func (e *Entity) GetUserTransaction(userDiscordId string) ([]model.OffchainTipBo
 func (e *Entity) GetTransactionsByQuery(senderId, receiverId, token string) ([]model.OffchainTipBotTransferHistory, error) {
 	return e.repo.OffchainTipBotTransferHistories.GetTransactionsByQuery(senderId, receiverId, token)
 }
+
+func (e *Entity) GetTransactionsByGuildIdAndToken(guildId, token, action string) ([]model.OffchainTipBotTransferHistory, error) {
+	tipbotActions := []string{"tip", "airdrop", "withdraw"}
+	if !util.Contains(tipbotActions, action) {
```
Comment (trkhoi) [inline pkg/entities/offchain_tip_bot_transfer_histories.go:18]: get all the action so no need this

File: pkg/repo/offchain_tip_bot_transfer_histories/pg.go [hunk 2]
```diff
@@ -64,8 +64,14 @@ func (pg *pg) GetTransactionsByQuery(senderId, receiverId, token string) ([]mode
...
-func (pg *pg) GetTotalTransactionByGuildAndToken(guildId, token string) (count int64, err error) {
-	return count, pg.db.Model(&model.OffchainTipBotTransferHistory{}).Where("guild_id = ? and token = ? and status = ?", guildId, strings.ToUpper(token), "success").Count(&count).Error
+func (pg *pg) GetTotalTransactionByGuildAndToken(guildId, token, action string) ([]model.OffchainTipBotTransferHistory, int64, error) {
+	var transferHistories []model.OffchainTipBotTransferHistory
+	var count int64
+	if action == "" {
+		//get all actions
+		return transferHistories, count, pg.db.Model(&model.OffchainTipBotTransferHistory{}).Where("guild_id = ? and token = ? and status = ?", guildId, strings.ToUpper(token), "success").Find(&transferHistories).Count(&count).Error
+	}
+	return transferHistories, count, pg.db.Model(&model.OffchainTipBotTransferHistory{}).Where("guild_id = ? and token = ? and action = ? and status = ?", guildId, strings.ToUpper(token), action, "success").Find(&transferHistories).Count(&count).Error
```
Comment (trkhoi) [inline pkg/repo/offchain_tip_bot_transfer_histories/pg.go:74]: do we need this anymore

File: pkg/entities/offchain_tip_bot_config_notify.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func (e *Entity) ListConfigNotify(guildId string) (rs []response.ConfigNotifyRes
 			}
 
 		} else {
-			totalTx, err = e.repo.OffchainTipBotTransferHistories.GetTotalTransactionByGuildAndToken(guildId, config.Token)
+			_, totalTx, err = e.repo.OffchainTipBotTransferHistories.GetTotalTransactionByGuildAndToken(guildId, config.Token, "")
```
Comment (trkhoi) [inline pkg/entities/offchain_tip_bot_config_notify.go:57]: why need another var if not used

**Review by trkhoi — CHANGES_REQUESTED** [review]

File: pkg/entities/offchain_tip_bot_config_notify.go (hunk 3 above)
Comment (catngh) [reply pkg/entities/offchain_tip_bot_config_notify.go]: this is from another module that only use the record count

**Review by hoangharry — APPROVED** [review]

**Pull request merged**